etrago/cluster/networkclustering.py

```diff
@@ -215,74 +337,57 @@ def cluster_on_extra_high_voltage(network, busmap, with_time=True):
+    )
+    io.import_components_from_dataframe(network_c, new_df, "Generator")
     for attr, df in iteritems(new_pnl):
-        io.import_series_from_dataframe(network_c, df, 'Generator', attr)
+        io.import_series_from_dataframe(network_c, df, "Generator", attr)
 
     # dealing with all other components
     aggregate_one_ports = network.one_port_components.copy()
-    aggregate_one_ports.discard('Generator')
+    aggregate_one_ports.discard("Generator")
 
     for one_port in aggregate_one_ports:
-        one_port_strategies = {'StorageUnit': {'marginal_cost': np.mean, 'capital_cost': np.mean,
-                             'efficiency_dispatch': np.mean, 'standing_loss': np.mean, 'efficiency_store': np.mean,
-                             'p_min_pu': np.min}, 'Store': {'marginal_cost': np.mean, 'capital_cost': np.mean,
-                             'standing_loss': np.mean, 'e_nom': np.sum}}
+        one_port_strategies = strategies_one_ports()
         new_df, new_pnl = aggregateoneport(
-            network, busmap, component=one_port, with_time=with_time,
-            custom_strategies=one_port_strategies.get(one_port, {}))
+            network,
+            busmap,
+            component=one_port,
+            with_time=with_time,
+            custom_strategies=one_port_strategies.get(one_port, {}),
+        )
         io.import_components_from_dataframe(network_c, new_df, one_port)
         for attr, df in iteritems(new_pnl):
             io.import_series_from_dataframe(network_c, df, one_port, attr)
-    
+
     # Dealing with links
     network2 = network.copy()
     network2.links.bus0 = network2.links.bus0.map(busmap)
     network2.links.bus1 = network2.links.bus1.map(busmap)
-    network2.links.dropna(subset= ['bus0', 'bus1'], inplace= True)
-    network2.links['topo'] = np.nan
-    
-    strategies={'scn_name': _make_consense_links,
-                'bus0': _make_consense_links,
-                'bus1': _make_consense_links,
-                'carrier': _make_consense_links,
-                'efficiency_fixed': _make_consense_links,
-                'p_nom': np.sum,
-                'p_nom_extendable': _make_consense_links,
-                'p_nom_max': np.sum,
-                'capital_cost': np.mean,
-                'length': np.mean,
-                'geom': nan_links,
-                'topo': nan_links,
-                'type': nan_links,
-                'efficiency': np.mean,
-                'p_nom_min': np.min,
-                'p_set': np.mean,
-                'p_min_pu':np.min,
-                'p_max_pu': np.max,
-                'marginal_cost': np.mean,
-                'terrain_factor': _make_consense_links,
-                'p_nom_opt': np.mean,
-                'country': _make_consense_links}
-    
-    network_c.links = network2.links.groupby(['bus0', 'bus1', 'carrier']).agg(strategies)
+    network2.links.dropna(subset=["bus0", "bus1"], inplace=True)
+    network2.links["topo"] = np.nan
```

Comment:
Some links (e.g. heat pumps) and lines have time series values. Is this considered in the clustering? I couldn't find it in your function.

Reply:
This problem will be fixed with issue #366